src/librustdoc/html/render/search_index.rs

```diff
@@ -226,11 +215,11 @@ fn get_index_type_name(clean_type: &clean::Type, accept_generic: bool) -> Option
             let path = &bounds[0].trait_;
             Some(path.segments.last().unwrap().name)
         }
-        clean::Generic(s) if accept_generic => Some(s),
+        clean::Generic(s) => Some(s),
         clean::Primitive(ref p) => Some(p.as_sym()),
-        clean::BorrowedRef { ref type_, .. } => get_index_type_name(type_, accept_generic),
-        clean::Generic(_)
-        | clean::BareFunction(_)
+        clean::BorrowedRef { ref type_, .. } => get_index_type_name(type_),
+
```

Comment:
nit: You added an empty line here.

Reply:
Yeah, that was intentional. It separates the variants that are handled from the ones that are ignored. I can remove it if you want.

Reply:
Yes please.